src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java

```diff
@@ -109,28 +110,32 @@ private void executeScriptChunk(
       String scriptName,
       Integer chunkNumber)
       throws SQLException, IOException {
-    String firstRowStamp = getUtcTimeStringFromTimestamp(resultSet.getTimestamp(labelColumn));
+    Timestamp previousTimestamp = new Timestamp(0);
```

Comment:
We can not rely on timestamps for split rows. Documentation doesn't guarantee that https://docs.teradata.com/r/Teradata-Database-Data-Dictionary/June-2017/Views-Reference/QryLogSQLV – it says `when DBQL SQL cache was written`. 
Here is an example of such SQL log with different timestamps for the same query ID:

```sql
SELECT CollectTimeStamp, QueryID FROM QryLogSQLV WHERE QueryID = 3071939516
48781243;

 *** Query completed. 2 rows found. 2 columns returned.
 *** Total elapsed time was 1 second.

          CollectTimeStamp               QueryID
--------------------------  --------------------
2022-02-13 16:36:11.240000    307193951648781243
2022-02-13 16:36:13.760000    307193951648781243
```

Reply:
We could rely on QueryID, but still there is a chance when running with timerange boundaries that one of the logs will fall out of these boundraries...

Reply:
I should've explained more that this change matters the most for the resumable/incremental runs. They (will, according to the current design) rely on last-saved-timestamps as the "checkpoint" for new runs, whose beginning timestamp of the new query's time range will be the last-saved-timestamp + 1us (Teradata's timestamp resolution). Thus it's important to make sure rows with the same sorting timestamps do not get separated into different chunks.